rwatch/ui/animation/property_animation.c

```diff
@@ -0,0 +1,172 @@
+    if (property_animation->impl.accessors.setter.grect)
+        property_animation->impl.accessors.setter.grect(layer, new_rect);
+    //layer_set_frame(layer, new_rect);
+    layer_mark_dirty(layer);
+}
+
+void property_animation_update_gpoint(PropertyAnimation * property_animation, const uint32_t distance_normalized)
+{
+    int progress = (distance_normalized * 100) / ANIMATION_NORMALIZED_MAX;
+    
+    Layer *layer = (Layer *) property_animation->subject;
+    
+    GRect *from = &property_animation->values.from.grect;
+    GRect *to = &property_animation->values.to.grect;
+    
+    GRect new_rect = *from;
+    new_rect.origin.x = from->origin.x + (((to->origin.x - from->origin.x) / 100.0)*progress);
+    new_rect.origin.y = from->origin.y + (((to->origin.y - from->origin.y) / 100.0)*progress);
+    if (property_animation->impl.accessors.setter.gpoint)
+        property_animation->impl.accessors.setter.gpoint(layer, new_rect.origin);
+    //layer_set_frame(layer, new_rect);
+    layer_mark_dirty(layer);
+}
+
+void property_animation_update_int16(PropertyAnimation * property_animation, const uint32_t distance_normalized)
+{
+    int progress = (distance_normalized * 100) / ANIMATION_NORMALIZED_MAX;
+    int16_t *int16 = (int16_t *) property_animation->subject;
+    
+    int16_t *from = &property_animation->values.from.int16;
+    int16_t *to = &property_animation->values.to.int16;
+    
+    property_animation->impl.accessors.setter.int16(int16, (int)(((to - from) / 100.0)*progress));
+}
+
+void property_animation_update_uint32(PropertyAnimation * property_animation, const uint32_t distance_normalized)
+{
+    int progress = (distance_normalized * 100) / ANIMATION_NORMALIZED_MAX;
+    uint32_t *uint32 = (uint32_t *) property_animation->subject;
+    
+    uint32_t *from = &property_animation->values.from.uint32;
+    uint32_t *to = &property_animation->values.to.uint32;
+    
+    property_animation->impl.accessors.setter.uint32(uint32, (int)(((to - from) / 100.0)*progress));
+}
+
+void property_animation_update_gcolor8(PropertyAnimation * property_animation, const uint32_t distance_normalized)
+{
+    int progress = (distance_normalized * 100) / ANIMATION_NORMALIZED_MAX;
+    
+    GColor8 *gcolor = (GColor8 *) property_animation->subject;
+    
+    GColor8 *from = &property_animation->values.from.gcolor;
+    GColor8 *to = &property_animation->values.to.gcolor;
+    
+    GColor8 new_gcolor = *from;
+    new_gcolor.r = from->r + (((to->r - from->r) / 100.0)*progress);
+    new_gcolor.g = from->g + (((to->g - from->g) / 100.0)*progress);
+    new_gcolor.b = from->b + (((to->g - from->b) / 100.0)*progress);
+    new_gcolor.a = from->a + (((to->a - from->a) / 100.0)*progress);
+    if (property_animation->impl.accessors.setter.gcolor)
```

Comment:
One thing of note -- perhaps we can save all of this computation entirely by returning early if the `setter` is null?